node/pkg/driver/node_test.go

```diff
@@ -895,23 +896,136 @@ func TestNodeUnpublishVolume(t *testing.T) {
 }
 
 func TestNodeGetVolumeStats(t *testing.T) {
+	volId := "someStorageType:vol-test"
+	volumePath := "/test/path"
+	stagingTargetPath := "/staging/test/path"
+	volumePathWithHostPrefix := GetPodPath(volumePath)
+	mockCtl := gomock.NewController(t)
+	defer mockCtl.Finish()
+	mockNodeUtils := mocks.NewMockNodeUtilsInterface(mockCtl)
+	d := newTestNodeService(mockNodeUtils, nil, nil)
 
-	req := &csi.NodeGetVolumeStatsRequest{}
+	testCases := []struct {
+		name     string
+		testFunc func(t *testing.T)
+	}{
+		{
+			name: "fail volumePath does not exists",
+			testFunc: func(t *testing.T) {
+				expErrCode := codes.NotFound
+				mockNodeUtils.EXPECT().GetPodPath(volumePath).Return(volumePathWithHostPrefix)
+				mockNodeUtils.EXPECT().IsPathExists(volumePathWithHostPrefix).Return(false)
+				req := &csi.NodeGetVolumeStatsRequest{
+					VolumeId:          volId,
+					VolumePath:        volumePath,
+					StagingTargetPath: stagingTargetPath,
+				}
 
-	d := newTestNodeService(nil, nil, nil)
+				_, err := d.NodeGetVolumeStats(context.TODO(), req)
+				assertError(t, err, expErrCode)
+			},
+		},
+		{
+			name: "fail to get stats",
+			testFunc: func(t *testing.T) {
+				expErrCode := codes.Internal
+				mockNodeUtils.EXPECT().GetPodPath(volumePath).Return(volumePathWithHostPrefix)
+				mockNodeUtils.EXPECT().IsPathExists(volumePathWithHostPrefix).Return(true)
+				mockNodeUtils.EXPECT().IsBlock(volumePathWithHostPrefix).Return(false, nil)
+				mockNodeUtils.EXPECT().IsNotMountPoint(volumePathWithHostPrefix).Return(false, nil)
+				mockNodeUtils.EXPECT().IsDirectory(volumePathWithHostPrefix).Return(true)
+				mockNodeUtils.EXPECT().GetVolumeStats(volumePathWithHostPrefix, true).Return(driver.VolumeStatistics{}, status.Errorf(codes.Internal, "fail to get stats"))
+				req := &csi.NodeGetVolumeStatsRequest{
+					VolumeId:          volId,
+					VolumePath:        volumePath,
+					StagingTargetPath: stagingTargetPath,
+				}
```

Comment:
it seems that this can be moved to the beginning of `TestNodeGetVolumeStats`

Reply:
done